src/Shared/CommunicationsUtilities.cs

```diff
@@ -122,7 +125,21 @@ private static int FileVersionHash
                     // them, so just check COMPLUS_InstallRoot.
                     string complusInstallRoot = Environment.GetEnvironmentVariable("COMPLUS_INSTALLROOT");
 
-                    s_fileVersionHash = GetHandshakeHashCode(complusInstallRoot ?? ThisAssembly.AssemblyInformationalVersion);
+#if THISASSEMBLY
+                    var fileIdentity = ThisAssembly.AssemblyInformationalVersion;
+#else
+                    var fileIdentity = string.Empty;
+
+                    using (var sha1 = SHA1.Create())
+                    {
+                        var hashBytes = sha1.ComputeHash(File.ReadAllBytes(AssemblyUtilities.GetAssemblyLocation(typeof(CommunicationsUtilities).Assembly)));
+                        fileIdentity = Encoding.UTF8.GetString(hashBytes);
+                    }
+
+                    ErrorUtilities.VerifyThrow(!string.IsNullOrEmpty(fileIdentity), "file hashing failed");
```

Comment:
This could end up decreasing though right? This doesn't seem great :(

Reply:
This is just getting a hash, not a version, though, right? Or do I misunderstand your concern?

Reply:
Discussed offline, hash is fine because it is only used in the handshake and nothing else.